test/doc/playground_test.lua

```diff
@@ -0,0 +1,128 @@
+-- output with reference return values.
+--
+-- The first arguments is the request string. All the following
+-- arguments are expected return values (as Lua values).
+--
+-- The function ignores trailing `null` values in the YAML
+-- output.
+local function check_request(request, ...)
+    local ph, err = popen.new({'./doc/playground.lua'}, {
+        stdin = popen.opts.PIPE,
+        stdout = popen.opts.PIPE,
+        stderr = popen.opts.DEVNULL,
+    })
+    if ph == nil then
+        error('popen.new: ' .. tostring(err))
+    end
+
+    local ok, err = ph:write(request, {timeout = 1})
+    if not ok then
+        ph:close()
+        error('ph:write: ' .. tostring(err))
+    end
+    ph:shutdown({stdin = true})
+
+    -- Read everything until EOF.
+    local chunks = {}
+    while true do
+        local chunk, err = ph:read()
+        if chunk == nil then
+            ph:close()
+            error('ph:read: ' .. tostring(err))
+        end
+        if chunk == '' then break end -- EOF
+        table.insert(chunks, chunk)
+    end
+
+    local status = ph:wait()
+    assert(status.state == popen.state.EXITED)
+
+    -- Glue all chunks, parse response.
+    local stdout = table.concat(chunks)
+    local response_yaml = string.match(stdout, '%-%-%-.-%.%.%.')
+    local response = yaml.decode(response_yaml)
+
+    -- NB: This call does NOT differentiate `nil` and `box.NULL`.
+    t.assert_equals(response, {...})
+end
+
+local cases = {
+    {
+        'test_select_customers',
+        request = "crud.select('customers', {{'<=', 'age', 35}})",
+        retval_1 = {
+            metadata = {
+                {name = 'id', type = 'unsigned'},
+                {name = 'bucket_id', type = 'unsigned'},
+                {name = 'name', type = 'string'},
+                {name = 'age', type = 'number'},
+            },
+            rows = {
+                {5, 1172, 'Jack', 35},
+                {3, 2804, 'David', 33},
+                {6, 1064, 'William', 25},
+                {7, 693, 'Elizabeth', 18},
+                {1, 477, 'Elizabeth', 12},
+            },
+        }
+    },
+    {
+        'test_select_developers',
+        request = "crud.select('developers', nil, {first = 6})",
+        retval_1 = {
+            metadata = {
+                {name = 'id', type = 'unsigned'},
+                {name = 'bucket_id', type = 'unsigned'},
+                {name = 'name', type = 'string'},
+                {name = 'surname', type = 'string'},
+                {name = 'age', type = 'number'},
+            },
+            rows = {
+                {1, 477, 'Alexey', 'Adams', 20},
+                {2, 401, 'Sergey', 'Allred', 21},
+                {3, 2804, 'Pavel', 'Adams', 27},
+                {4, 1161, 'Mikhail', 'Liston', 51},
+                {5, 1172, 'Dmitry', 'Jacobi', 16},
+                {6, 1064, 'Alexey', 'Sidorov', 31},
+            },
+        },
+    },
+    {
+        'test_insert',
```

Comment:
Names can be taken from keys, but it's really not critical:
```Lua
test_insert = {
    ...
},
```
[see an example](https://github.com/tarantool/crud/blob/20023a9a6faab1976997534e11c3b71f24d9061f/test/unit/stats_test.lua#L93-L209).

Reply:
It was my variant of declarative test case description, which I use, when the order was important. Here it is not the case, so I'll rewrite it as you suggested.